doc/sphinx/proof-engine/tactics.rst

```diff
@@ -3406,129 +3406,139 @@ Automation
    introduces the newly generated hypotheses as hints. Then it looks at
    the list of tactics associated to the head symbol of the goal and
    tries to apply one of them (starting from the tactics with lower
    cost). This process is recursively applied to the generated subgoals.
 
-   By default, auto only uses the hypotheses of the current goal and the
-   hints of the database named core.
+   By default, :tacn:`auto` only uses the hypotheses of the current goal and
+   the hints of the database named ``core``.
 
-.. tacv:: auto @num
+   .. warning::
 
-   Forces the search depth to be :token:`num`. The maximal search depth
-   is 5 by default.
+      :tacn:`auto` uses a weaker version of :tacn:`apply` that is closer to
+      :tacn:`simple apply` so it is expected that sometimes :tacn:`auto` will
+      fail even if applying manually one of the hints would succeed.
 
-.. tacv:: auto with {+ @ident}
+   .. tacv:: auto @num
 
-   Uses the hint databases :n:`{+ @ident}` in addition to the database core.
+      Forces the search depth to be :token:`num`. The maximal search depth
+      is 5 by default.
+
+   .. tacv:: auto with {+ @ident}
+
+      Uses the hint databases :n:`{+ @ident}` in addition to the database ``core``.
+
+      .. note::
+
+         Use the fake database `nocore` if you want to *not* use the `core`
+         database.
+
+   .. tacv:: auto with *
+
+      Uses all existing hint databases. Using this variant is highly discouraged
+      in finished scripts since it is both slower and less robust than the variant
+      where the required databases are explicitly listed.
 
    .. seealso::
       :ref:`The Hints Databases for auto and eauto <thehintsdatabasesforautoandeauto>` for the list of
       pre-defined databases and the way to create or extend a database.
 
-.. tacv:: auto with *
+   .. tacv:: auto using {+ @ident__i} {? with {+ @ident } }
 
-   Uses all existing hint databases.
+      Uses lemmas :n:`@ident__i` in addition to hints. If :n:`@ident` is an
+      inductive type, it is the collection of its constructors which are added
+      as hints.
 
-   .. seealso:: :ref:`The Hints Databases for auto and eauto <thehintsdatabasesforautoandeauto>`
+      .. note::
 
-.. tacv:: auto using {+ @ident__i} {? with {+ @ident } }
+         The hints passed through the `using` clause are used in the same
+         way as if they were passed through a hint database. Consequently,
+         they use a weaker version of :tacn:`apply` and :n:`auto using @ident`
+         may fail where :n:`apply @ident` succeeds.
 
-   Uses lemmas :n:`@ident__i` in addition to hints. If :n:`@ident` is an
-   inductive type, it is the collection of its constructors which are added
-   as hints.
+         Given that this can be seen as counter-intuitive, it could be useful
+         to have an option to use full-blown :tacn:`apply` for lemmas passed
+         through the `using` clause. Contributions welcome!
```

Comment:
Maybe we should have a wishlist somewhere?

Reply:
That's my attempt to start such a list. My take is that these wishes must be closed to the documentation of the corresponding component so that people can find them.